src/mds/MDSRank.cc

```diff
@@ -845,6 +845,18 @@ bool MDSRank::is_stale_message(Message *m) const
   return false;
 }
 
+Session *MDSRank::get_session(Message *m)
+{
+  Session *session = static_cast<Session *>(m->get_connection()->get_priv());
+  if (session) {
+    dout(20) << "get_session have " << session << " " << session->info.inst
+	     << " state " << session->get_state_name() << dendl;
+    session->put();  // not carry ref
```

Comment:
How about converting to a `boost::intrusive_ptr` instead of decrementing the ref here (which is a hack)? We could have a `SessionRef` instead of `Session *` that we pass around.

Reply:
That requires big change, not suite for backport